mne/beamformer/_lcmv.py

```diff
@@ -103,51 +121,74 @@ def _apply_lcmv(data, info, tmin, forward, noise_cov, data_cov, reg,
 
         # Find source orientation maximizing output source power
         if pick_ori == 'max-power':
-            eig_vals, eig_vecs = linalg.eigh(Ck)
-
-            # Choosing the eigenvector associated with the middle eigenvalue.
-            # The middle and not the minimal eigenvalue is used because MEG is
-            # insensitive to one (radial) of the three dipole orientations and
-            # therefore the smallest eigenvalue reflects mostly noise.
-            for i in range(3):
-                if i != eig_vals.argmax() and i != eig_vals.argmin():
-                    idx_middle = i
-
-            # TODO: The eigenvector associated with the smallest eigenvalue
-            # should probably be used when using combined EEG and MEG data
-            max_ori = eig_vecs[:, idx_middle]
-
-            Wk[:] = np.dot(max_ori, Wk)
-            Ck = np.dot(max_ori, np.dot(Ck, max_ori))
-            is_free_ori = False
-
-        if is_free_ori:
-            # Free source orientation
-            Wk[:] = np.dot(linalg.pinv(Ck, 0.1), Wk)
-        else:
-            # Fixed source orientation
-            Wk /= Ck
+            if weight_norm:  
+                # finding optimal orientation for NAI and unit-noise gain 
+                # based on Sekihara & Nagarajan 2008, Eq. 4.47            
+                tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))                
+                eig_vals, eig_vecs = linalg.eig(np.dot(linalg.pinv(tmp),np.dot(Wk,Gk)))
+                eig_vals = np.real(eig_vals)
+                idx_max = eig_vals.argmax()
+                max_ori = eig_vecs[:, idx_max]            
+                Wk[:] = np.dot(max_ori, Wk)    
+                Gk = np.dot(Gk, max_ori)
+                
+                if weight_norm == 'nai':
+                    tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
+                    denom = np.sqrt(noise * tmp)
+                    Wk /= denom
+                elif weight_norm == 'unitnoisegain':
+                    raise ValueError('not implemented yet.')                    
+                    
+                is_free_ori = False    
+                
+            else:
+                eig_vals, eig_vecs = linalg.eigh(Ck)
+
+                # Choosing the eigenvector associated with the middle eigenvalue.
+                # The middle and not the minimal eigenvalue is used because MEG is
+                # insensitive to one (radial) of the three dipole orientations and
+                # therefore the smallest eigenvalue reflects mostly noise.
+                for i in range(3):
+                    if i != eig_vals.argmax() and i != eig_vals.argmin():
+                        idx_middle = i
```

Comment:
another small thing, we try to use double letter local variables. ii, for example. Some of the letters are rerserved as special commands in the python debugger.